library/src/main/java/com/pokegoapi/api/gym/Battle.java

```diff
@@ -126,7 +122,7 @@ public Result start() throws LoginFailedException, RemoteServerException {
 	 */
 	public AttackGymResponse attack(int times) throws LoginFailedException, RemoteServerException {
 
-		ArrayList<BattleAction> actions = new ArrayList<BattleAction>();
+		ArrayList<BattleAction> actions = new ArrayList<>();
```

Comment:
Diamonds are supported since Java 7. That's the same Version gralde 3.0 needs to build.